cpp/src/slice2cs/Gen.cpp

```diff
@@ -1340,30 +1335,17 @@ Slice::Gen::TypesVisitor::visitClassDefEnd(const ClassDefPtr& p)
     // protected internal constructor used for unmarshaling (always generated).
     _out << sp;
     _out << nl << "protected internal " << name << "(global::Ice.InputStream istr, bool mostDerived)";
-    if (hasPublicParameterlessCtor)
-    {
-        // We call this() to get the field initialization (of the entire class hierarchy) + the call to Initialize.
-        _out.inc();
-        _out << nl << ": this()";
-        _out.dec();
-    }
-    else if (hasBaseClass)
+    if (hasBaseClass)
     {
         // We call the base class constructor to initialize the base class fields.
         _out.inc();
         _out << nl << ": base(istr, mostDerived: false)";
         _out.dec();
     }
     _out << sb;
-    if (!hasPublicParameterlessCtor) // i.e. we did not call this()
-    {
-        // We perform the initialization of defaulted members (and call Initialize) in this constructor.
-        writeDataMemberInitializers(dataMembers, ns, ObjectType, true);
-        if(partialInitialize)
-        {
-            _out << nl << "Initialize();";
-        }
-    }
+
+    // This initialization is only for tagged data members; for other data members, we are about to read them all.
+    writeDataMemberInitializers(dataMembers, ns, ObjectType, true);
```

Comment:
Is this comment accurate?  this is also about setting references to `null!` to avoid warnings